overlord/snapshotstate/snapshotstate.go

```diff
@@ -515,8 +535,11 @@ func Check(st *state.State, setID uint64, snapNames []string, users []string) (s
 // Forget creates a taskset for deletinig a snapshot.
 // Note that the state must be locked by the caller.
 func Forget(st *state.State, setID uint64, snapNames []string) (snapsFound []string, ts *state.TaskSet, err error) {
-	// forget needs to conflict with check and restore
-	if err := checkSnapshotTaskConflict(st, setID, "check-snapshot", "restore-snapshot"); err != nil {
+	// forget needs to conflict with check, restore, import and export.
+	// XXX: we could probably drop this since conflict check is now done inside
+	// doForget task handler.
```

Comment:
I don't understand the intention behind the suggested change,  before we construct the change is the right place to check for a conflict

Reply:
Yes yes, you're right; I confused myself by adding conflict check in actual handler which doesn't make sense since it's prevented precisely here. Thanks.